python/cudf/tests/test_joining.py

```diff
@@ -335,3 +335,69 @@ def test_dataframe_empty_merge():
     got = gdf1.merge(gdf2, how='left', on=['a'])
 
     assert_eq(expect, got)
+
+
+def test_dataframe_merge_order():
+    gdf1 = DataFrame()
+    gdf2 = DataFrame()
+    gdf1['id'] = [10, 11]
+    gdf1['timestamp'] = [1, 2]
+    gdf1['a'] = [3, 4]
+
+    gdf2['id'] = [4, 5]
+    gdf2['a'] = [7, 8]
+
+    gdf = gdf1.merge(gdf2, how='left', on=['id', 'a'], method='hash')
+
+    df1 = pd.DataFrame()
+    df2 = pd.DataFrame()
+    df1['id'] = [10, 11]
+    df1['timestamp'] = [1, 2]
+    df1['a'] = [3, 4]
+
+    df2['id'] = [4, 5]
+    df2['a'] = [7, 8]
+
+    df = df1.merge(df2, how='left', on=['id', 'a'])
+    assert_eq(gdf, df)
+
+
+@pytest.mark.parametrize('pairs', [('', ''), ('', 'a'), ('', 'ab'), ('', 'abc'), ('', 'b'), ('', 'bcd'), ('', 'cde'), ('a', 'a'), ('a', 'ab'), ('a', 'abc'), ('a', 'b'), ('a', 'bcd'), ('a', 'cde'), ('ab', 'ab'), ('ab', 'abc'), ('ab', 'b'), ('ab', 'bcd'), ('ab', 'cde'), ('abc', 'abc'), ('abc', 'b'), ('abc', 'bcd'), ('abc', 'cde'), ('b', 'b'), ('b', 'bcd'), ('b', 'cde'), ('bcd', 'bcd'), ('bcd', 'cde'), ('cde', 'cde')])   # noqa: E501
+@pytest.mark.parametrize('max', [5, 1000])
+@pytest.mark.parametrize('rows', [5, 15, 100])
```

Comment:
It looks like the `rows=0` case is a completely different column order from pandas when `right` is a subset of `left`. I don't think we should worry about mirroring pandas perfectly in this case, since reordering the columns of the output in any way is a no-op.